File: src/com/gmail/marcosav2010/communicator/packet/wrapper/OverExceededByteLimitException.java
package com.gmail.marcosav2010.communicator.packet.wrapper;

import com.gmail.marcosav2010.communicator.packet.AbstractPacket;

public class OverExceededByteLimitException extends RuntimeException {
    
	private static final long serialVersionUID = -3543010566122316736L;

	public OverExceededByteLimitException(long incoming) {
        super("Exceded max byte array length: " + AbstractPacket.MAX_SIZE + " < " + incoming + ".");
    }
}

File: core/src/main/java/com/gmail/marcosav2010/communicator/packet/wrapper/exception/OverExceededByteLimitException.kt
package com.gmail.marcosav2010.communicator.packet.wrapper.exception

import com.gmail.marcosav2010.communicator.packet.AbstractPacket

class OverExceededByteLimitException(incoming: Long) :
        RuntimeException("Exceeded max byte array length: ${AbstractPacket.MAX_SIZE} < $incoming.") {

    companion object {
        private const val serialVersionUID = -3543010566122316736L
    }
}